apps/cow-tools/src/hooks/useUserMilkmanTransactions.ts

```diff
@@ -0,0 +1,171 @@
+        swaps {
+          id
+          chainId
+          transactionHash
+          tokenAmountIn
+          priceChecker
+          orderContract
+          priceCheckerData
+          tokenIn {
+            id
+            name
+            symbol
+            decimals
+          }
+          tokenOut {
+            id
+            name
+            symbol
+            decimals
+          }
+        }
+      }
+    }
+  }
+`);
+
+export interface ICowOrder {
+  sellToken: Address;
+  buyToken: Address;
+  receiver: Address;
+  sellAmount: number;
+  buyAmount: number;
+  validTo: number;
+  feeAmount: number;
+  kind: string;
+  partiallyFillable: boolean;
+  sellTokenBalance: string;
+  buyTokenBalance: string;
+  from: Address;
+  executedSellAmount?: number;
+  executedSellAmountBeforeFees?: number;
+  executedBuyAmount?: number;
+  executedFeeAmount?: number;
+  status: string;
+}
+
+const cowApiUrl = "https://api.cow.fi/goerli";
+
+export function useUserMilkmanTransactions() {
+  const { safe } = useSafeAppsSDK();
+  const [loaded, setLoaded] = useState(false);
+  const [transactions, setTransactions] =
+    useState<AllTransactionFromUserQuery["users"][0]["transactions"]>();
+  const [cowOrders, setOrders] = useState<ICowOrder[][][]>();
+  const [hasToken, setHasToken] = useState<boolean[][]>();
+
+  const publicClient = createPublicClient({
+    chain: goerli,
+    transport: http(),
+  });
+
+  useEffect(() => {
+    async function loadOrders() {
+      const { users } = await sdk.AllTransactionFromUser({
+        user: safe.safeAddress,
+      });
+
+      const swapsLenByTransaction = users[0]?.transactions.map(
+        (transaction) => transaction.swaps.length,
+      );
+
+      const orderContractsByTransaction = users[0]?.transactions.map(
+        (transaction) => transaction.swaps.map((swap) => swap.orderContract),
+      );
+
+      const orderContractsBySwap = ([] as string[]).concat(
+        ...orderContractsByTransaction,
+      );
+
+      const tokenAddressesByTransactions = users[0]?.transactions.map(
+        (transaction) => transaction.swaps.map((swap) => swap.tokenIn?.id),
+      ) as (Address | undefined)[][];
+
+      const tokenAddressesBySwap = ([] as (Address | undefined)[]).concat(
+        ...tokenAddressesByTransactions,
+      );
+
+      const tokenBalances = (await Promise.all(
+        orderContractsBySwap.map((orderContract, index) => {
+          if (!tokenAddressesBySwap[index]) {
+            return 0;
+          }
+          return getTokenBalance(
+            tokenAddressesBySwap[index] as Address,
+            orderContract as Address,
+            publicClient,
+          );
+        }),
+      )) as number[];
+
+      const hasTokenBySwap = tokenBalances.map(
+        (tokenBalance) => tokenBalance > 0,
+      );
+
+      const cowOrdersBySwap = await Promise.all(
+        orderContractsBySwap.map((orderContract) =>
+          getCowOrders(orderContract as Address),
+        ),
+      );
```

Comment:
here is where we check if the transaction is in fact from milkman?

Reply:
Actually, here we check if one order was posted on cow. I can add some comments on these parts.